src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -153,6 +153,7 @@ void ColumnVsValueTableScanImpl::_scan_sorted_segment(const BaseSegment& segment
                                                          boost::get<ColumnDataType>(value));
 
         sorted_segment_search.scan_sorted_segment([&](auto begin, auto end) {
+          if (begin == end) return;
```

Comment:
(Not related to this fix)

It took me a while to understand how `scan_sorted_segment` handles NotEquals. While that behavior is documented in `sorted_segment_search.hpp`, I think a comment "In case of NotEquals, this is called twice" would have helped a lot.

Reply:
I will add that

Reply:
Hm, I don't see how it is called twice. The method `_handle_not_equals` determines two ranges and combines them. Is that what you mean?

Reply:
Yes - for me, that was quite unexpected.